book/machinelearning.qmd

```diff
@@ -4,114 +4,215 @@ abstract: TODO (150-200 WORDS)
+Returning to the `rats` data, consider a split that results in only male rats in the training data, the model would then be unable to handle female rats in the testing data, which it had never encountered before.
+Alternatively, consider a random split with an even sex balance, as only two out of the 150 rats have a tumor, there is a high probability that the training data will not include any male rat with a tumor.
+Hence, in the case of a male rat, the model is likely to ignore the treatment variable and instead just use sex as the most important predictor of tumor incidence.
+
+To deal with this problem, $k$-fold cross-validation (CV) can be used as a more robust method and to better estimate the generalization error [@Hastie2001].
+$k$-fold CV partitions the data into $k$ subsets, called *folds*.
+The training data comprises of $k-1$ of the folds and the remaining one is used for testing and evaluation.
+This is repeated $k$ times until each of the folds has been used exactly once as the testing data.
+The performance from each fold is averaged into a final performance estimate.
+It is common to use $k = 5$ or $k = 10$ [@Breiman1992; @Kohavi1995].
+This process can be repeated multiple times (*repeated $k$-fold CV*) and/or $k$ can even be set to $n$, which is known as *leave-one-out cross-validation*.
+
+Cross-validation can also be stratified, which ensures that a variable of interest will have the same distribution in each fold as in the original data.
+This is important, and often recommended, in survival analysis to ensure that the proportion of censoring in each fold is representative of the full dataset.
+<!-- FIXME: CITE OUR SURVIVAL BENCHMARK PAPER -->
+
+Repeating resampling experiments with multiple models is referred to as a *benchmark experiment*.
+A benchmark experiment compares models by evaluating their performance on *identical* data, which means the same resampling strategy and folds should be used for all models.
+Determining if one model is actually better than another is a surprisingly complex topic [@Demsar2006; @Dietterich1998; @Nadeau2003] and is out of scope for this book, instead any benchmark experiments performed in this book are purely for illustrative reasons and no results are expected to generalize outside of these experiments.
+
+## Optimization {#sec-ml-opt}
+
+@sec-ml-models introduced model hyperparameters, which control how training and prediction algorithms are run.
+Setting hyperparameters is a critical part of model fitting and can significantly change model performance.
+*Tuning* is the process of using internal benchmark experiment to automatically select the optimal hyper-parameter configuration.
+For example, the number of trees, $m_r$ in a random forest (@sec-surv-ml-models-ranfor) is a potential hyperparameter to tune.
```

Comment:
```suggestion
For example, the number of trees, $m_r$, in a random forest (@sec-surv-ml-models-ranfor) is a potential hyperparameter to tune.
```

Reply:
my least favorite example for hyperparam in RF :-) as theory says it should be as high as necessary, but higher in general better